### PR TITLE
Make secret volume plugin idempotent

### PR DESCRIPTION
**Pull request by pmorie**
@thockin @erictune

### CODE REVIEW

File: pkg/volume/secret/secret.go [hunk 1]
```diff
@@ -91,6 +92,10 @@ var wrappedVolumeSpec = &volume.Spec{
 }
 
 func (sv *secretVolume) SetUpAt(dir string) error {
+	if volumeutil.IsReady(sv.GetPath()) {
```
Comment (thockin) [inline pkg/volume/secret/secret.go]: This is going to create a file in the path that containers can access.  Is that really what you intended?  What if I create a secret named "ready"?

**Comment by thockin** [comment]
LGTM but the one question - I suspect you want to mimic gitrepo here

**Comment by pmorie** [comment]
@thockin I see on closer inspection that the git repo plugin uses the
PodPluginDir to store the ready file, which isn't exposed to the user.
Will refactor to use that.

**Comment by pmorie** [comment]
@thockin The ready file isn't exposed to the user now.

**Pull request merged**